### ISSUE TITLE
cc: error: /usr/local/lib/libixp.a: No such file or directory

### ISSUE THREAD

**Issue by dbb** [sunaku/wmii#1]
I'm on Debian testing. I installed the build deps as described in the readme:

```
libx11-dev:
  Installed: 2:1.6.3-1
  Candidate: 2:1.6.3-1
  Version table:
 *** 2:1.6.3-1 0
        990 http://ftp.us.debian.org/debian/ testing/main amd64 Packages
        300 http://ftp.us.debian.org/debian/ sid/main amd64 Packages
        100 /var/lib/dpkg/status

libixp-dev:
  Installed: 0.6~20121202+hg148-2
  Candidate: 0.6~20121202+hg148-2
  Version table:
 *** 0.6~20121202+hg148-2 0
        990 http://ftp.us.debian.org/debian/ testing/main amd64 Packages
        300 http://ftp.us.debian.org/debian/ sid/main amd64 Packages
        100 /var/lib/dpkg/status
```

But when I run `make` it fails with the following error:

```
MAKE all cmd/
FILTER cmd/wihack.sh
FILTER cmd/wmii.rc.rc
FILTER cmd/wmii.sh.sh
CC cmd/wmiir.o
LD cmd/wmiir.out
cc -o wmiir.out wmiir.o -L../lib -L/usr/lib ../lib/libstuff.a ../lib/libregexp9.a ../lib/libbio.a ../lib/libfmt.a ../lib/libutf.a /usr/local/lib/libixp.a -g -ldl
cc: error: /usr/local/lib/libixp.a: No such file or directory
../mk/hdr.mk:75: recipe for target 'wmiir.out' failed
make[1]: *** [wmiir.out] Error 1
mk/dir.mk:15: recipe for target 'dall' failed
make: *** [dall] Error 2
```

Is this because the `PREFIX` is set to `/usr/local` but the deps are installed to `/usr/lib` rather than `/usr/local/lib`? I didn't change the prefix because the readme said that no further config should be necessary on Debian, and in general I don't like to use `/usr` as a prefix when building from source.

If that's the problem, what's the recommended solution? Is it possible to simply copy the necessary libs over to `/usr/local` or is there a better method?

**Comment by sunaku** [comment]
You need to build and install [libixp](https://github.com/sunaku/libixp) _first_ before attempting to build wmii. :sweat_smile:

Also, It's fine to change the `PREFIX` value, inside the `config.mk` files, as far as I'm concerned. :smiling_imp:  I've always been installing of them (_first_ libixp, then wmii) to `~/app/wmii` because I lack root access:

``` sh
%?master ~/lab/libixp> git diff
diff --git a/config.mk b/config.mk
index 45ae851..4433f10 100644
--- a/config.mk
+++ b/config.mk
@@ -5,7 +5,7 @@ COMPONENTS = \
    libixp_pthread

 # Paths
-PREFIX = /usr/local
+PREFIX = $(HOME)/app/wmii
   BIN = $(PREFIX)/bin
   MAN = $(PREFIX)/share/man
   ETC = $(PREFIX)/etc
%?master ~/lab/libixp> cd ../wmii
%?master ~/lab/wmii> git diff
diff --git a/config.mk b/config.mk
index 0512ce7..11efc50 100644
--- a/config.mk
+++ b/config.mk
@@ -1,7 +1,7 @@
 # Customize below to fit your system

 # Paths
-PREFIX = /usr/local
+PREFIX = $(HOME)/app/wmii
   BIN = $(PREFIX)/bin
   MAN = $(PREFIX)/share/man
   DOC = $(PREFIX)/share/doc/wmii
```

**Comment by dbb** [comment]
Libixp is installed, per the aptitude output I posted. 

The problem is that make can't find it because of the prefix. The only prefix that would work in this scenario is `/usr` since `/usr//lib/libixp.a` exists, but it is my understanding that setting the prefix to `/usr` is a bad idea on debian because files that make installs there may conflict with existing files installed by apt. 

That is why I asked if it would be better to just copy the needed files from `/usr` to `/usr/local`. Rebuilding and reinstalling something that I already have installed just to put it in a different directory doesn't make much sense to me.

**Comment by sunaku** [comment]
Oh sorry, I didn't notice the `libixp-dev` package in your output. :sweat_smile: AFAIK the versions of libixp and wmii available in Debian have traditionally been out of date (even back when wmii itself was still popular). :older_man:

Coming to the present, I stopped using wmii over 3 years ago :skull:, so I really don't have any meaningful suggestions to address your dilemma other than the meager example (showing how I install both libixp and wmii to my local home folder) I had supplied earlier. :disappointed:   Good luck! :sparkles:
